pandas/tests/indexes/numeric/test_numeric.py

```diff
@@ -223,11 +223,14 @@ def test_fillna_float64(self):
 
     def test_logical_compat(self, dtype):
         idx = Index(np.arange(5, dtype=dtype))
-        assert idx.all() == idx.values.all()
-        assert idx.any() == idx.values.any()
+        msg = "cannot perform all with this index type: Index"
 
-        assert idx.all() == idx.to_series().all()
-        assert idx.any() == idx.to_series().any()
+        with pytest.raises(TypeError, match=msg):
+            assert idx.all() == idx.values.all()
```

Comment:
i think we only need the `idx.all()` call here, not the `assert` or `== [...]`. Should probably be separate pytest.raises calls to make sure each thing that raises does

Reply:
I rolled my changes back, because after removing `!= "m"` this test passes.